src/traces/scatter3d/calc_errors.js

```diff
@@ -19,9 +19,10 @@ function calculateAxisErrors(data, params, scaleFactor) {
     for(var i = 0; i < data.length; i++) {
         var errors = computeError(+data[i], i);
 
+        var point = axis.d2l(data[i], 0, calendar) * scaleFactor; // A bit wasteful
         result[i] = [
-            -errors[0] * scaleFactor,
-            errors[1] * scaleFactor
+            (axis.d2l(data[i] - errors[0], 0, calendar) * scaleFactor) - point || -point,
+            (axis.d2l(data[i] + errors[1], 0, calendar) * scaleFactor) - point || -point
```

Comment:
Since you brought `calendar` in here I'm assuming we're now starting to think about all axis types (including date and category), not just numeric (linear & log). At that level of generality, `d` formats cannot be added/subtracted, but `c` formats can. In fact since the errors are relative values, and for non-numeric axis types there isn't even really a concept of "zero" (see https://github.com/plotly/plotly.js/pull/2960#discussion_r215098710 for example) errorbar values should be assumed to be in `c` format (so milliseconds or number of categories)

Of course `+data[i]` from the `computeError` call above *also* won't work for these axis types, but that's actually a good thing, because it will prevent `'fraction'` and `'sqrt'` errorbars on these axes so you can only use `constant` or `data`. (hmm... except in pathological cases like when you have `'2018'` as a date or a bunch of numbers as "categories"... that's going to be a bug, we should really not try to do `+data[i]` on non-numeric axes and just pass `undefined` or something)

As you point out, all those `d2l` calls will be a bit of a drag... I suspect it'll be best to make a special case for log axes and leave the other types as they were.

Reply:
As you suggested, I now make a special case for log axes and leave the other types as they were.